src/main/groovy/com/github/lookout/whoas/RedisQueue.groovy

```
@@ -0,0 +1,138 @@
+package com.github.lookout.whoas
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import redis.clients.jedis.Jedis;
+
+/**
+ * A redis queue that offers distributed and persistent queue
+ */
+class RedisQueue extends AbstractHookQueue {
+    private final RedisClientFactory redisClientFactory
+    private static nextId = 0
+    private String hostname
+    private Integer port
+
+    /**
+     * Create the RedisQueue with valid config
+     */
+    RedisQueue(WhoasQueueConfig queueConfig) {
+        redisClientFactory = RedisClientFactory.getInstance()
+        this.hostname = queueConfig.hostname
+        this.port = queueConfig.port
+    }
+
+    /**
+     * Default constructor
+     */
+    RedisQueue() {
+        redisClientFactory = RedisClientFactory.getInstance()
+        this.hostname = "localhost"
+        this.port = 6379
+    }
+    /**
+     * Return the number of elements in the queue
+     */
+    Long getSize() {
+        if (!this.started) {
+            throw new Exception("Queue must be started before this operation is invoked")
+        }
+        Jedis redisclient = null
+        Long queueSize = 0
+        try {
+            redisclient = redisClientFactory.getJedis()
+            queueSize = redisclient.llen("queue")
+        } catch (all) {
```

This kind of code is really not okay, like, ever. Catching swallowing exceptions means that nobody will ever know what happens on errors here. If the `llen()` call throws we won't know, we'll just return zero, which incorrect. I would rather let all our exceptions from Jedis be thrown further up. Catching and wrapping in Whoas speciifc exceptions I don't think would provide any value either, but just serve as a middle-man exception class which hides information from a developer